pages/tutorials/[slug].js

```diff
@@ -30,10 +33,70 @@ export async function getStaticProps(context) {
 }
 
 export default function Tutorial({ name, steps, id, slug }) {
+  const [currentStep, setCurrentStep] = useState(0);
+
   return (
     <>
       <Headline>{name}</Headline>
-      <TutorialStartCard />
+      {currentStep === 0 ? (
```

Comment:
maybe you could also make a query here not only for currentStep===0, but also for currentStep<= steps.legth and get rid of l70-76?

Reply:
Thanks for challenging me, Mark! 😄 
I'm not sure if the code has become more readable but I think the new structure is easier to understand. Or what do you think?